tests/filter_test.php

```diff
@@ -87,7 +87,7 @@ public function test_empty_image() {
         ];
         foreach ($sizes as $size) {
             $emptyimage = phpunit_util::call_internal_method($filter, 'empty_image', $size, get_class($filter));
-            $this->assertContains('width="'.$size[0].'" height="'.$size[1].'"', $emptyimage);
+            $this->assertStringContainsString('width="'.$size[0].'" height="'.$size[1].'"', $emptyimage);
```

Comment:
PHP 7.4 retired December 6 2021, PHP 8.0 is much stricter.

Otherwise you get a TypeError: PHPUnit\Framework\Assert::assertContains(): Argument $haystack must be of type iterable, string given.

Reply:
Thanks for the explanation, that makes total sense.